contribution_plan/services.py

```diff
@@ -223,6 +224,12 @@ def delete(self, contribution_plan_bundle_details):
             return _output_exception(model_name="ContributionPlanBundleDetails", method="delete", exception=exc)
 
 
+def get_contribution_length(contribution_plan):
+    length = contribution_plan.periodicity
+    signal_result = get_contribution_length_signal.send(sender=ContributionPlanModel, contribution_plan=contribution_plan)[0][1]
```

Comment:
Here I would add in the CP model
length :float = None

then get_contribution_length will the a CP method
self.length = self.periodicity
get_contribution_length_signal.send(sender=self.__class__, instance=self) # not sure instance is required
return self.length

Reply:
you were not using signal_result